browser/ui/views/brave_actions/brave_actions_container.cc

```diff
@@ -135,22 +137,25 @@ void BraveActionsContainer::Init() {
   SetLayoutManager(std::move(vertical_container_layout));
 
   // children
+  onion_location_view_ = new OnionLocationView(browser_->profile());
```

Comment:
shouldn't we check the tor group policy or something before adding this?

Reply:
No, `BraveActionsContainer::Init()` is only called once per window creation.
And the pref toggle for `tor::TorProfileService::IsTorDisabled()` doesn't need restarting browser. So if we check and user toggle the pref from off to on, the label will not be shown until next relaunch.
This view will remain invisible until it has onion location and we should do cleanup as you mentioned in https://github.com/brave/brave-core/pull/6762#discussion_r501618579
Besides, chromium does the same thing in `PageActionIcon` like zoom, find...etc. Constructing first and then set visibility according to the condition.